### PR TITLE
feat(perf): improve performance of conversation filter API

### PR DESCRIPTION
**Pull request by scmmishra**
## Description

This PR fixes some N+1 queries for the conversation filter service. This caches the labels and precomputes the total message count. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

#### Before

CleanShot 2023-05-04 at 13 42 20@2x

#### After

CleanShot 2023-05-04 at 13 41 09@2x

### CODE REVIEW

File: app/views/api/v1/conversations/partials/_conversation.json.jbuilder [hunk 1]
```diff
@@ -17,7 +17,7 @@ json.meta do
 end
 
 json.id conversation.display_id
-if conversation.messages.count.zero?
+if conversation.message_count.zero?
   json.messages []
 elsif conversation.unread_incoming_messages.count.zero?
```
Comment (pranavrajs) [inline app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: @scmmishra Can we preload this as well?

Comment (scmmishra) [reply app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: I managed to do it, however in `models/messages.rb`, it is calculated again in `push_event_data` method. So the query is inevitable. Couldn't find a way to cache that

**Comment by scmmishra** [comment]
@sojan-official this is ready to go

File: app/views/api/v1/conversations/partials/_conversation.json.jbuilder (hunk 1 above)
Comment (sojan-official) [reply app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: We can avoid doing count here but rather go for conversation.messages.any? Instead. 
The count will trigger an additional count query, and tbh we don't need the specific value of count here anyways. 

https://twitter.com/nateberkopec/status/1654199212479860768?s=20 
https://www.speedshop.co/2019/01/10/three-activerecord-mistakes.html

File: app/services/conversations/filter_service.rb [hunk 2]
```diff
@@ -60,6 +60,12 @@ def conversations
     @conversations = @conversations.includes(
```
[the remark this answers is not in the document]
Comment (scmmishra) [reply app/services/conversations/filter_service.rb:60]: Not sure. I can try itout

[the remark this answers is not in the document]
Comment (sojan-official) [reply app/services/conversations/filter_service.rb:60]: we can pair on this.

**Comment by scmmishra** [comment]
Closing this PR, this history has too many unrelated changes. Raised a new PR here: https://github.com/chatwoot/chatwoot/pull/7120

**Pull request closed without merging**